examples/custom-server/server.js

```diff
@@ -0,0 +1,33 @@
+const { createServer } = require('http')
+const { parse } = require('url')
+const next = require('next')
+const accepts = require('accepts')
+
+const app = next('.', { dev: true })
+const handle = app.getRequestHandler()
+
+app.prepare()
+.then(() => {
+  createServer((req, res) => {
+    const accept = accepts(req)
+    const type = accept.type(['json', 'html'])
+    if (type === 'json') {
```

Comment:
Actually, my idea to remove this overhead from the user.

Anyway do we need to do this:

```
<Link href="/foo" as="/bar">
```

Since we are defining routes on the server. I think client doesn't need to know about anything about the actual mapping. So could do this:

```jsx
<Link href="/blog/postId">
```

If we can somehow do this, this will be awesome.